entity-framework/core/querying/user-defined-function-mapping.md

```diff
@@ -0,0 +1,154 @@
+---
+title: User-defined function mapping - EF Core
+description: Mapping user-defined functions to database functions
+author: maumar
+ms.date: 11/23/2020
+uid: core/modeling/custom-function-mapping
+---
+# User-defined function mapping
+
+EF Core allows for using user-defined SQL functions in queries. To do that, the functions need to be mapped to a CLR method during model configuration. When translating the LINQ query to SQL, the user-defined function is called instead of the CLR function it has been mapped to.
+
+## Mapping a method to a SQL function
+
+To illustrate how user-defined function mapping work, let's define the following entities:
+
+[!code-csharp[Main](../../../samples/core/Querying/UserDefinedFunctionMapping/Model.cs#BlogEntity)]
+
+[!code-csharp[Main](../../../samples/core/Querying/UserDefinedFunctionMapping/Model.cs#PostEntity)]
+
+[!code-csharp[Main](../../../samples/core/Querying/UserDefinedFunctionMapping/Model.cs#CommentEntity)]
+
+And the following model configuration:
+
+[!code-csharp[Main](../../../samples/core/Querying/UserDefinedFunctionMapping/Model.cs#EntityConfiguration)]
+
+Blog can have many posts, each post can be tagged with multiple tags, and each tag can be associated with multiple posts; we have set up a many-to-many relationship.
+
+Next, create the user-defined function `CommentedPostCountForBlog`, which returns the count of posts with at least one comment for a given blog, based on the blog `Id`:
+
+```sql
+CREATE FUNCTION dbo.CommentedPostCountForBlog(@id int)
+RETURNS int
+AS
+BEGIN
+    RETURN (SELECT COUNT(*)
+        FROM [Posts] AS [p]
+        WHERE ([p].[BlogId] = @id) AND ((
+            SELECT COUNT(*)
+            FROM [Comments] AS [c]
+            WHERE [p].[PostId] = [c].[PostId]) > 0));
+END
+```
+
+To use this function in EF Core, we define the following CLR method, which we map to the user-defined function:
+
+[!code-csharp[Main](../../../samples/core/Querying/UserDefinedFunctionMapping/Model.cs#BasicFunctionDefinition)]
+
+In the example, the method is defined on `DbContext`, but it can also be defined as a static method in other places. Note that the body of the CLR method is not important, as it will never be invoked client-side; EF Core only looks at the method signature.
```

Comment:
rather than other place - any other class.

Reply:
It can be invoked in client side if any of the argument cannot be translated and it is in projection. So never is somewhat strong word.

Reply:
Make static/instance a note.